src/mam4xx/wv_sat_methods.hpp

```diff
@@ -0,0 +1,137 @@
+  // the calculation for visibility's sake
+  // https://en.wikipedia.org/wiki/Goff-Gratch_equation
+  const Real svp_at_ice_pt_pressure = 6.1071; // BAD_CONSTANT!
+
+  return haero::pow(ten,
+                    -Real(9.09718) * (h2otrip / temperature - one) -
+                        Real(3.56654) * haero::log10(h2otrip / temperature) +
+                        Real(0.876793) * (one - temperature / h2otrip) +
+                        haero::log10(svp_at_ice_pt_pressure)) *
+         ten * ten;
+
+} // end GoffGratch_svp_ice
+
+// FIXME
+// Compute saturation vapor pressure over water
+KOKKOS_INLINE_FUNCTION
+Real svp_water(const Real temperature) {
+  // FIXME
+  // ask if we need to implement the other methods to compute svp_water
+  // initial_default_idx = GoffGratch_idx
+  return GoffGratch_svp_water(temperature);
+}
+
+/*---------------------------------------------------------------------
+! UTILITIES
+!---------------------------------------------------------------------*/
+
+// Get saturation specific humidity given pressure and SVP.
+// Specific humidity is limited to range 0-1.
+KOKKOS_INLINE_FUNCTION
+Real wv_sat_svp_to_qsat(const Real es, const Real p) {
+  // es  SVP
+  // p   Current pressure.
+  // If pressure is less than SVP, set qs to maximum of 1.
+  // epsilo  ! Ice-water transition range
+  // omeps   ! 1._r8 - epsilo
+  // epsilo       = shr_const_mwwv/shr_const_mwdair   ! ratio of h2o to dry air
+  // molecular weights real(R8),parameter :: SHR_CONST_MWDAIR  = 28.966_R8 !
+  // molecular weight dry air ~ kg/kmole real(R8),parameter :: SHR_CONST_MWWV
+  // = 18.016_R8       ! molecular weight water vapor
+  // const Real SHR_CONST_MWWV = 18.016;
+  // const Real SHR_CONST_MWDAIR = 28.966;
+  const Real epsilo = haero::Constants::weight_ratio_h2o_air;
+
+  const Real zero = 0;
+  const Real one = 1;
+  const Real omeps = one - epsilo;
+  Real qs = zero;
+  if ((p - es) <= zero) {
+    qs = one;
+  } else {
+    qs = epsilo * es / (p - omeps * es);
+  }
+  return qs;
+
+} // wv_sat_svp_to_qsat
+
+KOKKOS_INLINE_FUNCTION
+void wv_sat_qsat_water(const Real t, const Real p, Real &es, Real &qs) {
+  /*------------------------------------------------------------------!
+  ! Purpose:                                                         !
+  !   Calculate SVP over water at a given temperature, and then      !
+  !   calculate and return saturation specific humidity.             !
+  !------------------------------------------------------------------*/
+  // Inputs
+  // t    temperature
+  // p    Pressure
+  // Outputs
+  // es  Saturation vapor pressure
+  // qs  Saturation specific humidity
+
+  es = svp_water(t);
+  qs = wv_sat_svp_to_qsat(es, p);
+  // Ensures returned es is consistent with limiters on qs.
+  es = haero::min(es, p);
+
+} // wv_sat_qsat_water
+
+KOKKOS_INLINE_FUNCTION
+Real svp_ice(const Real temperature) { return GoffGratch_svp_ice(temperature); }
+
+} // namespace wv_sat_methods
+} // namespace mam4
+
```

Comment:
Note that most if not all of this is also implemented as part of SCREAM, we should check for consistency if we are implementing it again.

Reply:
if these functions are implemented in SCREAM, how should we invoke them in mam4xx ?

Reply:
For now, I think we should make a note of them. We don't have a mechanism to call external code like this, but if we wanted to support that, we could probably figure out how to add some smarts to the build system to allow it.

Reply:
Agreed @jeff-cohere. I just wanted to make a note for the record.